src/main/java/org/folio/dao/transactions/BatchTransactionPostgresDAO.java

```diff
@@ -0,0 +1,64 @@
+package org.folio.dao.transactions;
+
+import io.vertx.core.Future;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.rest.jaxrs.model.Transaction;
+import org.folio.rest.persist.Criteria.Criterion;
+import org.folio.rest.persist.CriterionBuilder;
+import org.folio.rest.persist.DBConn;
+import org.folio.rest.persist.interfaces.Results;
+
+import java.util.Collections;
+import java.util.List;
+
+public class BatchTransactionPostgresDAO implements BatchTransactionDAO {
+  private static final Logger logger = LogManager.getLogger();
+  public static final String TRANSACTIONS_TABLE = "transaction";
+
+  @Override
+  public Future<List<Transaction>> getTransactionsByCriterion(Criterion criterion, DBConn conn) {
+    logger.debug("Trying to get transactions by criterion: {}", criterion);
+    return conn.get(TRANSACTIONS_TABLE, Transaction.class, criterion)
+      .map(Results::getResults)
+      .onSuccess(transactions -> logger.info("Successfully retrieved {} transactions", transactions.size()))
+      .onFailure(e -> logger.error("Getting transactions by criterion failed, criterion: {}", criterion, e));
+  }
+
+  @Override
+  public Future<List<Transaction>> getTransactionsByIds(List<String> ids, DBConn conn) {
+    logger.debug("Trying to get transactions by ids = {}", ids);
+    if (ids.isEmpty()) {
+      return Future.succeededFuture(Collections.emptyList());
+    }
+    CriterionBuilder criterionBuilder = new CriterionBuilder("OR");
+    ids.forEach(id -> criterionBuilder.with("id", id));
```

Comment:
I suggest to extract these lines into a new method. There are a lot of duplication

Reply:
I added a method.